File: Array/Set Matrix Zeros.go
package main

import "fmt"

func main() {

	arr := [][]int{{1, 0, 1}, {1, 1, 1}, {1, 0, 1}}
	setZeroes(arr)
	fmt.Println(arr)
}

func setZeroes(matrix [][]int) {
	row := make([]bool, len(matrix))
	col := make([]bool, len(matrix[0]))

	for i := 0; i < len(matrix); i++ {
		for j := 0; j < len(matrix[0]); j++ {
			if matrix[i][j] == 0 {
				row[i] = true
				col[j] = true
			}
		}
	}

	for i := 0; i < len(matrix); i++ {
		for j := 0; j < len(matrix[0]); j++ {
			if row[i] == true || col[j] == true {
				matrix[i][j] = 0
			}
		}
	}
}

File: Pythons Solutions/Set Matrix Zeros.py
class Solution:
    def setToZero(self, arr):
        row, col = [False]*len(arr), [False]*len(arr[0])

        for i in range(len(arr)):
            for j in range(len(arr[0])):
                if arr[i][j] == 0:
                    row[i] = True
                    col[j] = True

        for i in range(len(arr)):
            for j in range(len(arr[0])):
                if row[i] == True or col[j] == True:
                    arr[i][j] == 0

        return arr

obj = Solution()
arr = [[1, 0, 0],[1, 1, 1],[1, 1, 1]]
ans = obj.setToZero(arr)
print(ans)